fred2/campaigneditordlg.cpp

```diff
@@ -139,11 +141,17 @@ void campaign_editor::OnLoad()
 		}
 	}
 
-	auto res = cf_find_file_location(Campaign.missions[Cur_campaign_mission].name, CF_TYPE_MISSIONS);
+	// try to open the file from the same folder as the campaign
+	if (!m_current_campaign_path.IsEmpty()) {
+		auto full_mission_path = GetPathWithoutFile();
+		full_mission_path.Append(Campaign.missions[Cur_campaign_mission].name);
 
-	if (res.found) {
-		FREDDoc_ptr->SetPathName(res.full_name.c_str());
+		auto res = cf_find_file_location((LPCTSTR)full_mission_path, CF_TYPE_MISSIONS, false);
+		if (res.found) {
+			FREDDoc_ptr->SetPathName(res.full_name.c_str());
+		}
 	}
+
```

Comment:
Is this correct? I looks like it will only look for files if the campaign path is set. Wasn't there a fallback here previously for when it's not set? I'm guessing that the addition of `GetPathWithoutFile()` made that obsolete, and the IsEmpty check was meant to be removed?

Reply:
Right, previously it would load from the regular mod data sources.  However, for whatever reason, retail missions stored in root_fs2.vp would not load properly.  It found them but couldn't read them.  I decided to restrict loading to only those cases where the mission was found on the file system, since that's really the only way FRED is used anyway.

Reply:
Ah, okay then.